lib/mcp/access_list_reviews_tools.go

```diff
@@ -0,0 +1,304 @@
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var result []accessListDetails
+	for _, list := range lists {
+		details, err := s.getAccessListDetails(ctx, list.GetName())
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		result = append(result, *details)
+	}
+	return result, nil
+}
+
+func (s *accessListsForReviewTool) getAccessListDetails(ctx context.Context, name string) (*accessListDetails, error) {
+	list, err := s.cfg.Auth.AccessListClient().GetAccessList(ctx, name)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	memberRoles, err := s.getAccessListRoles(ctx, list)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// TODO(r0mant): 200 members per list max should be fine for now, but we
+	// might want to paginate in the future.
+	members, _, err := s.cfg.Auth.AccessListClient().ListAccessListMembers(ctx, name, 200, "")
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// TODO(r0mant): similar to the above, 100 reviews should be plenty for now.
+	reviews, _, err := s.cfg.Auth.AccessListClient().ListAccessListReviews(ctx, name, 100, "")
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	url := ""
+	if s.cfg.WebProxyAddr != "" {
+		url = fmt.Sprintf(accessListURLTemplate, s.cfg.WebProxyAddr, name)
+	}
+
+	return &accessListDetails{
+		Name:        list.GetName(),
+		Description: list.GetMetadata().Description,
+		Labels:      list.GetMetadata().Labels,
+		Title:       list.Spec.Title,
+		Owners:      list.GetOwners(),
+		Audit:       list.Spec.Audit,
+		MemberRoles: memberRoles,
+		Members:     members,
+		Reviews:     reviews,
+		URL:         url,
+	}, nil
+}
+
+func (s *accessListsForReviewTool) getAccessListRoles(ctx context.Context, list *accesslist.AccessList) ([]*types.RoleV6, error) {
+	var memberRoles []*types.RoleV6
+	for _, roleName := range list.GetGrants().Roles {
+		role, err := s.cfg.Auth.GetRole(ctx, roleName)
+		if err != nil && !trace.IsAccessDenied(err) {
+			return nil, trace.Wrap(err)
+		}
+		// Owners may not have permissions to read roles.
+		if trace.IsAccessDenied(err) {
+			continue
+		}
+		roleV6, ok := role.(*types.RoleV6)
+		if !ok {
+			return nil, trace.BadParameter("expected role %q to be RoleV6, got %T", roleName, role)
+		}
+		memberRoles = append(memberRoles, roleV6)
+	}
+	return memberRoles, nil
+}
+
+// NewReviewAccessListTool creates an MCP tool that submits access list review.
+func NewReviewAccessListTool(cfg Config) (Tool, error) {
+	return &reviewAccessListTool{cfg}, nil
+}
+
+type reviewAccessListTool struct {
+	cfg Config
+}
+
+func (s *reviewAccessListTool) GetTool() mcp.Tool {
+	return mcp.NewTool(ToolReviewAccessList,
+		mcp.WithDescription(`Submit Teleport access lists review.`),
+		mcp.WithString("name", mcp.Required()))
+}
+
+func (s *reviewAccessListTool) GetHandler() server.ToolHandlerFunc {
+	return func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
```

Comment:
We are in the progress of migrating to official SDK (though it's currently blocked by
https://github.com/modelcontextprotocol/go-sdk/issues/657)

However, when we do, it can be tricky to convert this part. Official SDK use functions with typed input and output so it can properly convert them to input/output schema:
https://github.com/modelcontextprotocol/go-sdk/blob/411d5a049ae1006cf01ab9f5ac49d249ab3e8e2b/examples/server/everything/main.go#L183-L190

That said, we could worry about this later when we actually do the conversion.